core/runtime/v2/shim/util_unix.go

```diff
@@ -19,25 +19,32 @@
 package shim
 
 import (
+	"bufio"
 	"context"
 	"crypto/sha256"
 	"fmt"
+	"io"
+	"math"
 	"net"
 	"os"
 	"path/filepath"
 	"runtime"
+	"strconv"
 	"strings"
 	"syscall"
 	"time"
 
 	"github.com/containerd/containerd/v2/defaults"
 	"github.com/containerd/containerd/v2/pkg/namespaces"
 	"github.com/containerd/containerd/v2/pkg/sys"
+	"github.com/mdlayher/vsock"
 )
 
 const (
 	shimBinaryFormat = "containerd-shim-%s-%s"
 	socketPathLimit  = 106
+	Vsock            = "vsock"
+	HVsock           = "hvsock"
```

Comment:
Do we need to export?

Reply:
I'd also name this `hybridVsock` for clarity.

Reply:
done